src/cert_store.rs

```diff
@@ -243,6 +243,61 @@ impl CertStore {
             Ok(ret)
         }
     }
+
+    /// Determines if a cert exists in a given store
+    pub fn find_existing_cert(&mut self, cert: &CertContext) -> io::Result<CertContext> {
+        unsafe {
+            let res = Cryptography::CertFindCertificateInStore( 
+                self.0,
+                Cryptography::X509_ASN_ENCODING | Cryptography::PKCS_7_ASN_ENCODING,
+                0,                             
+                Cryptography::CERT_COMPARE_EXISTING << Cryptography::CERT_COMPARE_SHIFT,
+                cert.as_inner() as *const c_void, 
+                ptr::null_mut(), 
+            ); 
+            if res.is_null() { 
+                return Err(io::Error::last_os_error());
+            } else { 
+                Ok(CertContext::from_inner(res))
+            }  
+        } 
+    }
+
+    /// Searches provided store to see if the provided cert exisits.
+    /// If the cert already exists within the cert store, the newly added private key is deleted to get rid of duplicate keys
+    /// If the cert does not exist, the provided cert will be persisted to the store.
+    pub fn find_existing_cert_and_key(cert: &mut CertContext, store: &mut CertStore) -> io::Result<Option<CertContext>> {
+        let mut existing_cert = None; 
+
+        match store.find_existing_cert(cert) {
```

Comment:
A cleaner way instead of the match is to use "if let"

if let Ok(x) = store.find_existing_cert(cert) {
    if x
    }
}

Since we changed the match to an if let, you can remove the Err(_)

Reply:
This was a tad difficult to understand but I got it. Changed over.